### PR TITLE
Address some issues from code review

### PR DESCRIPTION
**Pull request by MonsieurNicolas**

### CODE REVIEW

File: src/transactions/TransactionFrame.cpp [hunk 1]
```diff
@@ -1626,16 +1653,15 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
...
             {
-                if (!isSoroban() && !mEvents.empty())
-                {
-                    throw std::runtime_error("unexpected events size");
-                }
-
-                outerMeta.pushContractEvents(std::move(mEvents));
-                outerMeta.pushDiagnosticEvents(std::move(mDiagnosticEvents));
-                outerMeta.setReturnValue(std::move(mReturnValue));
+                outerMeta.pushContractEvents(
```
Comment (MonsieurNicolas) [inline src/transactions/TransactionFrame.cpp:1659]: @sisuresh I think that this is the correct change (we should never set "SorobanData" for non Soroban transactions)

Comment (sisuresh) [reply src/transactions/TransactionFrame.cpp:1659]: Yeah good catch. This didn't matter when these fields lived under `TransactionMetaV3`, but now that they're under the optional `SorobanTransactionMeta`, it should only be activated for Soroban.

**Comment by MonsieurNicolas** [comment]
some of the issues fixed:
* make it impossible to reuse cached Soroban members in `TransactionFrame`
* do not set optional `SorobanMeta` in `LedgerCloseMeta` on non-soroban transactions
* error prone/bad test in herder (found while adding a few asserts) -> renamed `setFee` into `setFullFee` to better mirror getters
* add an explicit check for inconsistent fee values

some more minor things:
 * visual studio project had a bad profile
 * warnings/shadowed variables (tests for the most part)

File: src/transactions/InvokeHostFunctionOpFrame.cpp [hunk 2]
```diff
@@ -79,10 +70,11 @@ getLedgerInfo(AbstractLedgerTxn& ltx, Config const& cfg,
     info.cpu_cost_params = toCxxBuf(sorobanConfig.cpuCostParams());
     info.mem_cost_params = toCxxBuf(sorobanConfig.memCostParams());
     // TODO: move network id to config to not recompute hash
-    auto networkID = sha256(cfg.NETWORK_PASSPHRASE);
```
Comment (sisuresh) [inline src/transactions/InvokeHostFunctionOpFrame.cpp:82]: Nit: The comment above this ("// TODO: move network id to config to not recompute hash") can be removed.

File: src/transactions/InvokeHostFunctionOpFrame.cpp [hunk 3]
```diff
@@ -79,10 +70,11 @@ getLedgerInfo(AbstractLedgerTxn& ltx, Config const& cfg,
...
+    auto& networkID = app.getNetworkID();
```
Comment (sisuresh) [inline src/transactions/InvokeHostFunctionOpFrame.cpp:73]:
```suggestion
    auto const& networkID = app.getNetworkID();
```

File: Builds/VisualStudio/stellar-core.vcxproj [hunk 4]
```diff
@@ -268,7 +268,7 @@ exit /b 0
       </IgnoreSpecificDefaultLibraries>
     </Link>
     <PreBuildEvent>
-      <Command>(set CFLAGS=-MDd) &amp; (set CXXFLAGS=-MDd) &amp; cargo build --target-dir $(OutDir)\rust\target --features tracy</Command>
+      <Command>(set CFLAGS=-MDd) &amp; (set CXXFLAGS=-MDd) &amp; cargo build --target-dir $(OutDir)\rust\target --features tracy --features core-vnext</Command>
```
[the remark this answers is not in the document]
Comment (MonsieurNicolas) [reply Builds/VisualStudio/stellar-core.vcxproj:271]: they were already there, just this profile was missed

File: src/transactions/InvokeHostFunctionOpFrame.cpp (hunk 2 above)
Comment (MonsieurNicolas) [reply src/transactions/InvokeHostFunctionOpFrame.cpp:82]: fixed

File: src/transactions/InvokeHostFunctionOpFrame.cpp (hunk 3 above)
Comment (MonsieurNicolas) [reply src/transactions/InvokeHostFunctionOpFrame.cpp:73]: networkID is already const

**Comment by MonsieurNicolas** [comment]
addressed comments, rebased because of conflicts
also had to bump fmt and Visual Studio project files as an update that came out yesterday broke it

**Pull request merged**